pkg/scheduler/objects/object_state.go

```diff
@@ -78,12 +78,25 @@ func NewObjectState() *fsm.FSM {
 		},
 		fsm.Callbacks{
 			"enter_state": func(_ context.Context, event *fsm.Event) {
+				var object string
+				switch v := event.Args[0].(type) {
+				case *Queue:
+					object = v.QueuePath
+				case string:
+					object = v
```

Comment:
When is it "Queue" and when is it "string"? It seems to me that we only handle events from a single place so `event.Args[0]` will always be a "Queue" from this point on.

Reply:
The `PartitionContext` also uses the same state machine. It gives a partition name.

https://github.com/apache/yunikorn-core/blob/68a82820f0d7ff82f4d5d9b44629ae3660a184e1/pkg/scheduler/partition.go#L88

https://github.com/apache/yunikorn-core/blob/68a82820f0d7ff82f4d5d9b44629ae3660a184e1/pkg/scheduler/partition.go#L268

Reply:
Then simply introduce a new state machine for Queue.